src/mats_utils/geolocation/coordinates.py

```diff
@@ -178,4 +181,204 @@ def angles(ccditem):
     obs=sundir.altaz()
     TPsza = (90-obs[0].degrees)
     TPssa = (np.rad2deg(np.arccos(np.dot(FOV,sundir.position.m/norm(sundir.position.m)))))
-    return nadir_sza, TPsza, TPssa, tpLT
\ No newline at end of file
+    return nadir_sza, TPsza, TPssa, tpLT
+
+
+def pix_deg2(ccditem, xpixel, ypixel):
```

Comment:
so this could replace pix_deg - -  but we get it from level1_processing  we could put it there-

but would need to add the flipping bit

Reply:
So do I have to flip the image for the channels 1, 3, 5 and 6 ? 
There is no flipping at all for the channel 7 and the geolocated images are correctly oriented. (when I take L1a data)

Reply:
this is my orginal code
if (ccditem['CCDSEL']) in [1, 3, 5, 6]:
        xdeg = xdisp*((2048-ncskip - (ncol+1)*ncbin +
                      ncbin*(xpixel+0.5)) - 2047./2)
    else:
        xdeg = xdisp*(ncskip + ncbin * (xpixel+0.5) - 2047./2)
    ydeg = ydisp*(nrskip + nrbin * (ypixel+0.5) - 510./2)
    return xdeg, ydeg
you just need to add the if statement for the xdeg  and adjust it to use your f instead of crisp

Reply:
Should this take exposure time into account?

Reply:
I added the flipping but since I use an arctan function there is a difference of about 3e-3 degrees between my angles and the one in your code.
Since the images have been calibrated on the old version of the pix_deg function, should I just add a condition for calculating channel 7 with my method and leave the other channels unchanged?

Reply:
3e-3 is not a problem so no worries.  (not sure what you mean by "Since the images have been calibrated on the old version of the pix_deg function" - it is not used in the calibration stage I think - only when we plot . I think we can use your method for all .  

the best time to used for the geolocation is probably EXPDate + 'TEXPMS'/2*1000 . obs'TEXPMS' is in milliseconds